backend/wallets/handlers/wallet.py

```diff
@@ -0,0 +1,127 @@
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+from rest_framework.request import Request
+
+from django.http import HttpResponseNotAllowed
+
+from wallets.models import Wallet
+from wallets.controllers import WalletController
+from wallets.serializers.wallet import WalletCreateSerializer, WalletSerializer
+from wallets.utilities import HandlersChecks
+
+
+class WalletHandler:
+    """
+    Handler for the Wallet model.
+
+    This handler contains the views for the Wallet model.
+    It specifies the methods that should be used to perform actions on the Wallet model,
+    when a request is made to the API.
+
+    Attributes:
+        model_class: The Wallet model class.
+
+    Methods:
+        WalletView: A method that handles the GET and POST requests for the Wallet model.
+            It serves request without specific instance.
+        WalletDetailView: A method that handles the GET, PATCH and DELETE requests for the Wallet model.
+            It serves request with specific instance.
+
+    """
+
+    model_class = Wallet
+
+    @staticmethod
+    @api_view(['GET', 'POST'])
+    def WalletView(request: Request) -> Response:
+        """
+        Handle the GET and POST requests for the Wallet model.
+
+        Args:
+            request (Request): The request.
+
+        Returns:
+            Response: The response for user.
+        """
+        if request.method == 'GET':
+
+            wallets = WalletController.list_wallets(request)
+
+            return Response(WalletSerializer(wallets, many=True).data, status=200)
+            
+
+        elif request.method == 'POST':
+
+            serializer = WalletCreateSerializer(data=request.data, context={'request': request})
+
+            if serializer.is_valid():
+
+                try:
+                    wallet = WalletController.build_wallet(request.data, request.user)
+                    return Response(WalletCreateSerializer(wallet).data, status=201)
+                
+                except Exception as e:
+                    return Response({'error': str(e)}, status=500)
```

Comment:
Avoid using try/except for flow control, and almost always avoid capturing generic exceptions, since you can swallow errors and makes it difficult to understand what is happening. Only capture expected errors, if there is a 500, it will explode anyway

Reply:
True, I didn't think it through.